File: 3503.longest-palindrome-after-substring-concatenation-i.py
#
# @lc app=leetcode id=3503 lang=python3
#
# [3503] Longest Palindrome After Substring Concatenation I
#

# @lc code=start
class Solution:
    def longestPalindrome(self, s: str, t: str) -> int:
        def is_palindrome(sub: str) -> bool:
            return sub == sub[::-1]
        max_length = 0
        for i in range(len(s) + 1):
            for j in range(len(t) + 1):
                combined = s[:i] + t[:j]
                if is_palindrome(combined):
                    max_length = max(max_length, len(combined))
        return max_length
# @lc code=end

File: 3503.longest-palindrome-after-substring-concatenation-i.cpp
#
# @lc app=leetcode id=3503 lang=cpp
#
# [3503] Longest Palindrome After Substring Concatenation I
#

# @lc code=start
class Solution {
public:
    int longestPalindrome(string s, string t) {
        int maxLength = 0;
        auto isPalindrome = [](const string &str) {
            int left = 0, right = str.size() - 1;
            while (left < right) {
                if (str[left++] != str[right--]) return false;
            }
            return true;
        };
        for (int i = 0; i <= s.size(); ++i) {
            for (int j = 0; j <= t.size(); ++j) {
                string combined = s.substr(0, i) + t.substr(j);
                if (isPalindrome(combined)) {
                    maxLength = max(maxLength, int(combined.size()));
                }
            }
        }
        return maxLength;
    }
};
# @lc code=end